boilermaker/app.py

```diff
@@ -263,15 +282,20 @@ async def message_handler(
                 await self.publish_task(task.on_success)
         except RetryException as retry:
             # A retry has been requested:
-            # no on_failure run until after retries exhausted
+            # Calculate next delay and publish retry.
+            # Do not run on_failure run until after retries exhausted!
+            # The `retry` RetryException may have a policy -> What if it's different from the Task?
+            if retry.policy and retry.policy != task.policy:
+                # This will publish the *next* instance of the task using *this* policy
+                task.policy = retry.policy
+                logger.warning(f"Task policy updated to retry policy {retry.policy}")
+
             delay = task.get_next_delay()
             warn_msg = (
-                "Event retry requested. Publishing retry:"
-                f"[ {task.function_name=}"
-                f" {retry.msg=}"
-                f" {task.attempts.attempts=}"
-                f" {delay=}"
-                f" {sequence_number=} ]"
+                f"{retry.msg}. "
+                f"Retrying... <function={task.function_name}, "
+                f" attempts={task.attempts.attempts}, "
```

Comment:
Might be interesting to have the log message include the max retries. Something like `attempt = 2 of 5`

Reply:
ah that's a good point